### DISCUSSION TITLE
No matching distribution found for torch>=1.7 (from TTS) with Python 3.6 installation on Jetson Nano (aarch64)

### DISCUSSION THREAD

**Discussion by ScottMonaghan** [coqui-ai/STT#2064] [Q&A] [answered]
Hello,

I'm hoping to try out TTS synthesis on my Jetson Nano (aarch64).

I was hopeful that I'd get it working when I try to install for python3.6 I get the following error:
` Could not find a version that satisfies the requirement torch>=1.7 (from TTS) (from versions: )
No matching distribution found for torch>=1.7 (from TTS)`

And while it's true there's no official pytorch distribution for cp36-cp36m-linux_aarch64, wheels have been built and provided by Nvidia and I currently have torch 1.7 installed:
![image](https://user-images.githubusercontent.com/1868388/147802442-15ffc0ae-2c58-4ad8-bbe6-37a9619501a5.png)

Any ideas on how to get this to install on my Jetson Nano?

Here is the full output from the pip install:
`robud@robud:~/Downloads$ python3.6 -m pip install TTS
Collecting TTS
  Using cached https://files.pythonhosted.org/packages/44/70/31474245173e819c5a5def8ceb11277a9724067590b606e56619c868c6cd/TTS-0.4.2.tar.gz
Collecting cython (from TTS)
  Using cached https://files.pythonhosted.org/packages/15/29/2abb8975ded365d55b9e14129cabdfb977255911c80d8709028eca5829cd/Cython-0.29.26-py2.py3-none-any.whl
Collecting flask (from TTS)
  Using cached https://files.pythonhosted.org/packages/8f/b6/b4fdcb6d01ee20f9cfe81dcf9d3cd6c2f874b996f186f1c0b898c4a59c04/Flask-2.0.2-py3-none-any.whl
Collecting gdown (from TTS)
  Using cached https://files.pythonhosted.org/packages/54/e9/689f86828abda8413d7b39dec64a3614df90dbf29e80df8aea5207abca45/gdown-4.2.0.tar.gz
Collecting inflect (from TTS)
  Using cached https://files.pythonhosted.org/packages/4f/a8/031641ad73a1bd1a9932261a6193864556172b333dde263fed8b5a0940cf/inflect-5.3.0-py3-none-any.whl
Collecting jieba (from TTS)
  Using cached https://files.pythonhosted.org/packages/c6/cb/18eeb235f833b726522d7ebed54f2278ce28ba9438e3135ab0278d9792a2/jieba-0.42.1.tar.gz
Collecting librosa==0.8.0 (from TTS)
  Using cached https://files.pythonhosted.org/packages/26/4d/c22d8ca74ca2c13cd4ac430fa353954886104321877b65fa871939e78591/librosa-0.8.0.tar.gz
Collecting matplotlib (from TTS)
  Using cached https://files.pythonhosted.org/packages/22/d4/e7ca532e68a9357742604e1e4ae35d9c09a4a810de39a9d80402bd12f50f/matplotlib-3.3.4.tar.gz
Collecting numpy==1.19.5 (from TTS)
Collecting pandas (from TTS)
  Using cached https://files.pythonhosted.org/packages/fb/e4/828bb9c2474ff6016e5ce96a78220d485436d5468c23068f4f6c2eb9cff8/pandas-1.1.5.tar.gz
Collecting pypinyin (from TTS)
  Using cached https://files.pythonhosted.org/packages/1d/97/87bb66ff62fb1a9a6d216e31913778335a125be813597f3a70d88b077e0f/pypinyin-0.44.0-py2.py3-none-any.whl
Collecting pysbd (from TTS)
  Using cached https://files.pythonhosted.org/packages/48/0a/c99fb7d7e176f8b176ef19704a32e6a9c6aafdf19ef75a187f701fc15801/pysbd-0.3.4-py3-none-any.whl
Collecting pyyaml (from TTS)
  Using cached https://files.pythonhosted.org/packages/36/2b/61d51a2c4f25ef062ae3f74576b01638bebad5e045f747ff12643df63844/PyYAML-6.0.tar.gz
Collecting scipy>=0.19.0 (from TTS)
  Using cached https://files.pythonhosted.org/packages/aa/d5/dd06fe0e274e579e1dff21aa021219c039df40e39709fabe559faed072a5/scipy-1.5.4.tar.gz
Collecting soundfile (from TTS)
  Using cached https://files.pythonhosted.org/packages/eb/f2/3cbbbf3b96fb9fa91582c438b574cff3f45b29c772f94c400e2c99ef5db9/SoundFile-0.10.3.post1-py2.py3-none-any.whl
Collecting tensorboardX (from TTS)
  Using cached https://files.pythonhosted.org/packages/98/88/977b2f03fd0f8a2490fc7a1ad691d5e44cee5f1dc90c57078c5c168e2e70/tensorboardX-2.4.1-py2.py3-none-any.whl
Collecting torch>=1.7 (from TTS)
  Could not find a version that satisfies the requirement torch>=1.7 (from TTS) (from versions: )
No matching distribution found for torch>=1.7 (from TTS)`

**Comment by domcross** [comment, marked as answer]
Assuming you are installing into a Python virtual environment:
1) install Pytorch with CUDA support as provided by Nvidia: https://forums.developer.nvidia.com/t/pytorch-for-jetson-version-1-10-now-available/72048 - watch out for the torch-wheel that matches your JetPack version
2) pip install TTS

To my personal experience Jetson Nano is too slow for TTS, depending on the TTS-model synthesizing one second of audio can take up to 10 seconds...

**Comment by ScottMonaghan** [comment]
Ah thanks Dominik! I’m not surprised the Nano can’t handle it with the
quality if the result, but I wanted to confirm.

Your word is good enough for me! I’ll stick with my espeak-ng voice for my
robot:)

The STT, on the other hand seems to work well with tflite on the Nano.

On Fri, Dec 31, 2021 at 1:25 AM Dominik ***@***.***> wrote:

> Assuming you are installing into a Python virtual environment:
>
>    1. install Pytorch with CUDA support as provided by Nvidia:
>    https://forums.developer.nvidia.com/t/pytorch-for-jetson-version-1-10-now-available/72048
>    - watch out for the torch-wheel that matches your JetPack version
>    2. pip install TTS
>
> To my personal experience Jetson Nano is too slow for TTS, depending on
> the TTS-model synthesizing one second of audio can take up to 10 seconds...
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/coqui-ai/STT/discussions/2064#discussioncomment-1890571>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAOIEZHBQGIOAJVJV4Z5JODUTVLIJANCNFSM5LAXBYRQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
-- 
Scott Monaghan
413.244.9344
***@***.***
www.linkedin.com/in/scott-monaghan
Resume: https://tinyurl.com/yxhq5eul

**Reply by reuben** [reply]
For future reference, the TTS repo is https://github.com/coqui-ai/TTS
